integration-libs/cdc/root/service/cdc-js.service.spec.ts

```diff
@@ -412,25 +430,29 @@ describe('CdcJsService', () => {
     });
 
     it('should not do anything', () => {
-      spyOn(winRef.nativeWindow['gigya'].accounts, 'register');
+      spyOn(
+        winRef.nativeWindow && winRef.nativeWindow['gigya'].accounts,
+        'register'
+      );
       service['onInitRegistrationHandler']({}, null);
       expect(
-        winRef.nativeWindow['gigya'].accounts.register
+        winRef.nativeWindow && winRef.nativeWindow['gigya'].accounts.register
       ).not.toHaveBeenCalled();
     });
   });
 
   describe('loginUserWithoutScreenSet', () => {
     it('should login user without screenset', (done) => {
-      spyOn(winRef.nativeWindow['gigya'].accounts, 'login').and.callFake(
-        (options: { callback: Function }) => {
-          options.callback({ status: 'OK' });
-        }
-      );
+      spyOn(
+        winRef.nativeWindow && winRef.nativeWindow['gigya'].accounts,
+        'login'
+      ).and.callFake((options: { callback: Function }) => {
+        options.callback({ status: 'OK' });
+      });
       expect(service.loginUserWithoutScreenSet).toBeTruthy();
       service.loginUserWithoutScreenSet('uid', 'password').subscribe(() => {
         expect(
-          winRef.nativeWindow['gigya'].accounts.login
+          winRef.nativeWindow && winRef.nativeWindow['gigya'].accounts.login
```

Comment:
No need to change. Just a hint for shorter writing:
`winRef.nativeWindow?.['gigya'].accounts.login`

Reply:
noted, but VSCode lint was complaining even after I added the `?`. Will add it. thanks @Platonn